src/components/Api.js

```diff
@@ -0,0 +1,157 @@
+export default class Api {
+  constructor({ baseUrl, authToken }, id, avatar, isLiked) {
+    this._baseUrl = baseUrl;
+    this._authToken = authToken;
+    this._id = id;
+    this._avatar = avatar;
+    this._isLiked = isLiked;
+  }
+
+  getCardList() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: {
+        authorization: this._authToken,
+      },
+    }).then((res) =>
+      res.ok ? res.json() : Promise.reject(`Error: ${res.status}`)
+    );
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: {
+        authorization: this._authToken,
+      },
+    })
+      .then((res) =>
+        res.ok ? res.json() : Promise.reject(`Error: ${res.status}`)
+      )
+      .catch((err) => {
```

Comment:
I was told that every api needs a .catch, would I just use a finally instead of catch in those instances?

Reply:
> I was told that every api needs a .catch, would I just use a finally instead of catch in those instances?

`catch` is needed when you **call** the methods in `index.js` (not here)

And `finally` doesn't catch errors